baksnapperd.sh

```diff
@@ -38,6 +38,21 @@ function warning {
     echo -e "[Warning] $1" 1>&2
 }
 
+function link-latest {
+    arr=()
```

Comment:
Use `declare -a` as that will make the array local to the function.
Also try and use more descriptive names than `arr`, for example `snapshots`.